### ISSUE TITLE
Teams workflow support

### ISSUE THREAD

**Issue by zerocarbthirty** [sebbmeyer/php-microsoft-teams-connector#31]
Hello,

I began getting messages on my webhooks like this:

"Action Required:
O365 connectors within Teams will be deprecated and notifications from this service will stop.
[Learn more](https://aka.ms/O365ConnectorDeprecation)
about the timing and how the Workflows app provides a more flexible and secure experience. If you want to continue receiving these types of messages, you can use a workflow to post messages from a webhook request."

I just wanted to let you know that the new URLs don't work with this library.

**Comment by davehardyuk** [comment]
Yes, we are seeing this too, please can you let us know if there's any plans to upgrade this prior to the October 1st, 2024 retirement date?

**Comment by sebbmeyer** [comment]
Thanks for the information. I will take a look into it, if there is a way for migrating the current library to work with Power Automate workflows.

**Comment by zerocarbthirty** [comment]
Even if you could somehow only get the 'send a message from a webhook'
template to work that would pretty much do it. I'll see if I can get my
icinga2 plugin to work and if I can I can probably help make this work too.

**Comment by sebbmeyer** [comment]
Well we got a little more time for the adjustment, but I will keep it on my mind, and when I have some spare time, I will look into this issue

**Comment by davidkryzaniak** [comment]
@zerocarbthirty @sebbmeyer - I'm just quickly posting my findings on this _adventure_...

### Protips

**DO. NOT. USE. THE. TEMPLATE. AVAILABLE. IN. WORKFLOWS!** It doesn't work. I'm talking about this:

<img width="318" alt="image" src="https://github.com/user-attachments/assets/4fbbd6a3-1745-44d9-8604-ca680e2d6488">


Keep an eye on the URLs that Teams/Workflow gives to you. They should look like this: `https://prod-NN.XXXXus.logic.azure.com:443/workflows/***/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%2Ftriggers%2Fmanual%2Frun&sv=1.0&sig=***` Several times I've copied the URL, but only the "api-version" is included. Without the "sig", you will get a 401 error.

I'm unsure why, but I'm generally having better luck editing Workflows on https://make.powerautomate.com/ instead of inside Teams.

### Here's where I'm at

1. Create a new Automated cloud flow: 
![image](https://github.com/user-attachments/assets/bc45fc5a-e7f1-4258-b554-abefbf6ceace)
2. You will be asked to "Choose your flow's trigger". Click "skip".
3. Click "Add a trigger", search for "Teams Webhook".
![image](https://github.com/user-attachments/assets/6c56b004-62e0-4784-8d8d-2f0d11eac973)
4. Make sure "Who can trigger the flow?" is set to "Anyone".
![image](https://github.com/user-attachments/assets/69b90086-542b-44d7-b6b8-4b1bd9552d55)
5a. **[Deviating a bit here to just get anything to work as expected]** Add a "Parse JSON" action.
5b. Set the "Context" to "Body".
5c. Use `{"type":"object","properties":{"text":{"type": "string"}}}` as the Schema.
![image](https://github.com/user-attachments/assets/811589e4-0e5f-4f50-9126-8201cf343d54)
6a. Add a "Post message in a chat or channel".
6b. Set "Post As" to "Flow Bot".
6c. Set "Post In" to "Channel".
6d. Set a "Team".
6e. Set a "Channel".
6f. Set the "Message" to "Body text".
![image](https://github.com/user-attachments/assets/9a2b9252-f019-4a26-b49c-6d552b3662e1)
7. Save your Workflow.
8. Check out the "When a Teams webhook request is received" action, and grab the URL.
9. At this point your Workflow should look like this 
![image](https://github.com/user-attachments/assets/7788db4c-53df-42de-b4b0-81ba18f8afdf)
10. Add this to something: `<?php file_get_contents($THE_URL_FROM_WORKFLOWS', false, stream_context_create(['http'=>['header'=>"Content-Type: application/json\r\n",'method'=>'POST','content'=>json_encode(['text'=>'<h1>i like</h1><p>turtles</p>'])]])); ?>`
12. Hurray! You're still failing at the task at hand, but you're making progress!
<img width="1045" alt="image" src="https://github.com/user-attachments/assets/0fcef731-d216-439d-973d-ef8348303518">

This shouldn't be this difficult.

**Comment by ConstantinSimonis** [comment]
any news on this?

**Comment by Huydw** [comment]
No

**Comment by 4pack** [comment]
Well this teams connector module is officially broken with the discontinuation of Teams Connectors on January 30th. Has anyone found any alternatives?
